src/main/java/chess/domain/dao/ChessDao.java

```diff
@@ -0,0 +1,170 @@
+            Map<PositionDto, PieceDto> chessBoard = chessRoomDto.getChessBoard();
+            for (PositionDto positionDto : chessBoard.keySet()) {
+                pstmt2.setString(1, positionDto.getPosition());
+                pstmt2.setString(2, chessBoard.get(positionDto).getName());
+                pstmt2.setString(3, chessBoard.get(positionDto).getColor());
+                pstmt2.addBatch();
+                pstmt2.clearParameters();
+            }
+            pstmt2.executeBatch();
+            con.commit();
+            return 1;
+        }catch (Exception e) {
+            con.rollback();
+            return 0;
+        }finally {
+            closeConnection(con);
+        }
+    }
+
+    public ChessRoomDto findChessRoomByRoomNo(int roomNo) throws SQLException {
+        String query = "SELECT * FROM chess_room r JOIN chess_piece b ON r.room_no = b.room_no WHERE r.room_no = ?;";
+        PreparedStatement pstmt = getConnection().prepareStatement(query);
+        pstmt.setInt(1, roomNo);
+        ResultSet rs = pstmt.executeQuery();
+        validateRoomEmpty(rs);
+        rs.previous();
+        ChessRoomDto chessRoomDto = new ChessRoomDto();
+        Map<PositionDto, PieceDto> chessBoard = new HashMap<>();
+        while (rs.next()) {
+            PositionDto positionDto = new PositionDto(rs.getString("coordinate"));
+            PieceDto pieceDto = new PieceDto(rs.getString("piece_name"), rs.getString("piece_color"));
+            chessBoard.put(positionDto, pieceDto);
+            chessRoomDto.setRoom_no(rs.getInt("room_no"));
+            chessRoomDto.setBlackScore(rs.getDouble("black_score"));
+            chessRoomDto.setWhiteScore(rs.getDouble("white_score"));
+            chessRoomDto.setTurn(rs.getString("turn"));
+        }
+        chessRoomDto.setChessBoard(chessBoard);
+        return chessRoomDto;
+    }
+
+    private void validateRoomEmpty(ResultSet rs) throws SQLException {
+        if (!rs.next()) {
+            throw new IllegalArgumentException("[ERROR] 불러올 방이 없습니다.");
+        }
+    }
+
+    public int updateChessRoom(ChessRoomDto chessRoomDto, PieceDto targetPiece, PositionDto source, PositionDto target) throws SQLException {
+        Connection con = null;
+        try {
+            con = getConnection();
+            con.setAutoCommit(false);
+            String query1 = "UPDATE chess_room SET black_score = ?, white_score = ?, turn = ? where room_no = 1";
+            PreparedStatement pstmt1 = con.prepareStatement(query1);
+            pstmt1.setDouble(1, chessRoomDto.getBlackScore());
+            pstmt1.setDouble(2, chessRoomDto.getWhiteScore());
+            pstmt1.setString(3, chessRoomDto.getTurn());
+            pstmt1.executeUpdate();
+            updateChessPiece(targetPiece, source, target, con);
+            con.commit();
+            return 1;
+        }catch (Exception e) {
+            con.rollback();
+            return 0;
+        }finally {
+            closeConnection(con);
+        }
```

Comment:
```suggestion
        } catch (Exception e) {
            con.rollback();
            return 0;
        } finally {
            closeConnection(con);
        }
```

해당 부분은 전체적으로 확인 부탁 드릴게요 :)